### DISCUSSION TITLE
Add content to PDF right after existing content

### DISCUSSION THREAD

**Discussion by mibocs** [mpdf/mpdf#1788] [General]
How can i add new content to an PDF file(that i import) right after the existing content? So i wouldn't like to create new page and add new content there. Additionaly if the new content is more than one page then new pages should be automatically created. Is this even possible?

Example code:
`$autoloader = require_once 'autoload.php';

$mpdf = new \Mpdf\Mpdf();

$pagecount = $mpdf->setSourceFile('test.pdf');
$tplId = $mpdf->importPage($pagecount);
$mpdf->useTemplate($tplId);

$mpdf->WriteHTML('Hello World');

$mpdf->Output();`

This code just overrides the PDF.
![mpdf_pdf](https://user-images.githubusercontent.com/792039/210265156-ca11730e-5b52-4e91-a69e-000ba648902b.png)

Thanks

**Comment by TheArKaID** [comment]
Did you find solution for this?

**Reply by mibocs** [reply]
@TheArKaID Yes. We have to add pages to the mpdf object.

```
$autoloader = require_once 'autoload.php';

$mpdf = new \Mpdf\Mpdf();

$pagesInFile = $mpdf->setSourceFile('test.pdf');
        for ($i = 1; $i <= $pagesInFile; $i++) {
          $tplId = $mpdf->importPage($i);
          $size = $mpdf->getTemplateSize($tplId);

          // Create a page.
          // Landscape or portrait depending on the imported page size.
          if ($size['width'] > $size['height']) {
            $mpdf->AddPage(
              'L',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              0,
              0,
              0,
              0,
              '',
              [$size['width'], $size['height']]
            );
          }
          else {
            $mpdf->AddPage(
              'P',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              '',
              0,
              0,
              0,
              0,
              '',
              [$size['width'], $size['height']]
            );
          }

          $mpdf->useTemplate($tplId);
     }
$mpdf->SetX($mpdf->x);
$mpdf->SetY($mpdf->y);
$mpdf->WriteHTML('Hello World');

$mpdf->Output();
```

**Reply by finwe** [reply]
Hint: you can use [AddPageByArray()](https://mpdf.github.io/reference/mpdf-functions/addpagebyarray.html) with relevant keys only which will be much less verbose.

**Reply by TheArKaID** [reply]
Where did `$context['sandbox']` come from? Since the `WriteHtml` will place it based on the `SetX` and `SetY`

**Reply by mibocs** [reply]
@TheArKaID Ahh sorry, i fixed it.

**Reply by TheArKaID** [reply]
@mibocs  Just to make sure, is your `test.pdf` is an `mpdf` generated pdf? Or it come from like save as pdf from Doc(x)?

**Reply by mibocs** [reply]
@TheArKaID 

It doesn't matter if it is mpdf or doc(x) or anything else related.

**Reply by TheArKaID** [reply]
@mibocs  I've tried your code (literally copy and paste), and switched the `test.pdf` with my own, converted from Docx. And the `Hello World` printed in the top of the page like this

![image](https://github.com/mpdf/mpdf/assets/35050722/4d43928c-27e7-41a3-b1d2-c7292a0587c0)

The printed result of `x`, `y` and page sizes
![image](https://github.com/mpdf/mpdf/assets/35050722/7bccac51-b759-4624-bc7b-7527656fd577)

**Reply by mibocs** [reply]
@TheArKaID Try it without "$mpdf->SetX()" and "$mpdf->SetY".

**Reply by TheArKaID** [reply]
@mibocs still the same, the `Hello World` printed in the top.

Another trial for this, `mPDF` generated, and the `Hello World` still in the top

![image](https://github.com/mpdf/mpdf/assets/35050722/52467dd7-7319-4d89-afad-f281f297b817)

**Reply by TheArKaID** [reply]
@mibocs did your `Hello World` printed right after the last content in your pdf? Could you do some debugging for me like....what is the `x` and `y` before and after `writeHtml` based on your pdf?

**Reply by mibocs** [reply]
@TheArKaID I fixed "$pagesInFile" in the code. Try again.

**Reply by TheArKaID** [reply]
@mibocs  Actually I use the fixed version already, and the response is like before.

**Reply by mibocs** [reply]
@TheArKaID Then something is not alright with your mpdf library because i have just tested it and worked. The "Hello world" was on the last page, although the position of it was not fine.

**Reply by TheArKaID** [reply]
@mibocs  what do you mean by "the position of it was not fine."? Is it printed after the last content, or what? Also. which `mPDF` version did you use? 8.2?

**Reply by mibocs** [reply]
@TheArKaID As i said the "Hello world" had been on the last page, but it was not after the content but over the content, near to the end of the content. My mpdf version is 8.2.

**Reply by mibocs** [reply]
@TheArKaID
![image](https://github.com/mpdf/mpdf/assets/792039/7fb10f9e-bd73-45b9-86c3-58c55458c254)
